src/main/java/software/sigma/bu003/internship/coparts/controller/GlobalControllerExceptionHandler.java

```diff
@@ -0,0 +1,39 @@
+package software.sigma.bu003.internship.coparts.controller;
+
+import com.mongodb.MongoWriteException;
+import org.springframework.web.bind.annotation.ControllerAdvice;
+import org.springframework.web.bind.annotation.ExceptionHandler;
+import org.springframework.http.HttpStatus;
+import org.springframework.http.ResponseEntity;
+import org.springframework.web.bind.annotation.ResponseStatus;
+import software.sigma.bu003.internship.coparts.service.exception.PartNotFoundException;
+
+import javax.validation.ConstraintViolationException;
+
+@ControllerAdvice
+public class GlobalControllerExceptionHandler {
+
+    @ExceptionHandler(PartNotFoundException.class)
+    @ResponseStatus(HttpStatus.NOT_FOUND)
+    public ResponseEntity<String> partNotFoundException(PartNotFoundException e) {
+        return ResponseEntity
+                .status(HttpStatus.NOT_FOUND)
+                .body(e.getMessage());
+    }
+
+    @ExceptionHandler(MongoWriteException.class)
+    @ResponseStatus(HttpStatus.ACCEPTED)
+    public ResponseEntity<String> partAlreadyCreated() {
+        return ResponseEntity
+                .status(HttpStatus.ACCEPTED)
+                .body("This Entity already created.");
```

Comment:
MongoWriteException can happen not only when partAlreadyCreated.... will be wrong info

Reply:
Reworked this exception handling

Reply:
This is done for PartService.createPart(). When you save a duplicate entity, the repository throws a MongoWriteException. But I couldn't catch the MongoWriteException, it throws even in the try-catch block. I went all the way to the RuntimeException that was caught